### PR TITLE
[Feature Request]: Adding solution for 84. Largest Rectangle in Histogram

### PR DESCRIPTION
**Pull request by aaradhyasinghgaur**
## Related Issue

Fix: #634

## Description

I have added the leetcode problem number 82 solution in different languages with proper description , example , intuition , approach , proper clean codes , time and space complexity with conclusion.

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [x] Other (specify): Adding lc solution for 84

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

**Comment by kyra-09** [comment]
@Ajay-Dhangar kindly review my pr and merge it if there is no issue.

**Comment by kyra-09** [comment]
@Ajay-Dhangar could you please review my pr and in case of no problems could you please merge it?

File: dsa-solutions/lc-solutions/0000-0099/0084-largest-rectangle-in-histogram.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+#### Example 2
+
+- **Input:** $heights = [2,4]$
+- **Output:** $4$
+
+### Constraints
+
+
+- 1 <= heights.length <= 105
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0000-0099/0084-largest-rectangle-in-histogram.md:37]: **Bad practice**

```md
- 1 <= heights.length <= 105
- 0 <= heights[i] <= 104
```

**Good Practice**

```md
- $1 <= heights.length <= 105$
- $0 <= heights[i] <= 104$
```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Now, please update your code based on the suggestions provided and resolve any errors that occurred with my Docker setup.

File: dsa-solutions/lc-solutions/0000-0099/0084-largest-rectangle-in-histogram.md (hunk 1 above)
Comment (aaradhyasinghgaur) [reply dsa-solutions/lc-solutions/0000-0099/0084-largest-rectangle-in-histogram.md:37]: Alright... I'm new to this setup.. I'll resolve the issue ASAP.

**Comment by Ajay-Dhangar** [comment]
@aaradhyasinghgaur could you please update it? Otherwise, I may need to close it.

**Comment by aaradhyasinghgaur** [comment]
Yes.. I'm doing it right now

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**